hydpy/models/lland/lland_control.py

```diff
@@ -880,41 +920,50 @@ class KapGrenz(parametertools.Parameter):
     CONTROLPARAMETERS = (WMax, FK)
     NDIM, TYPE, TIME, SPAN = 2, float, None, (None, None)
     INIT = 0.0
+    KEYWORDS = {"option": parametertools.Keyword(name="option")}
 
     def __call__(self, *args, **kwargs) -> None:
-        try:
+        self._keywordarguments = parametertools.KeywordArguments(False)
+        idx = self._find_kwargscombination(args, kwargs, ({"option"},))
+        if idx is None:
             super().__call__(*args, **kwargs)
-        except NotImplementedError:
-            if len(kwargs) > 1:
-                raise ValueError(
-                    f"Parameter {objecttools.elementphrase(self)} does not "
-                    f"accept multiple keyword arguments, but the following "
-                    f"are given: {objecttools.enumeration(kwargs.keys())}"
-                ) from None
-            if "option" not in kwargs:
-                raise ValueError(
-                    f"Besides the standard keyword arguments, parameter "
-                    f"{objecttools.elementphrase(self)} does only support "
-                    f"the keyword argument `option`, but `{tuple(kwargs)[0]}` "
-                    f"is given."
-                ) from None
-            con = self.subpars
-            self.values = 0.0
-            if kwargs["option"] == "FK/2_FK":
-                self.values[:, 0] = 0.5 * con.fk  # type: ignore[index]
-                self.values[:, 1] = con.fk  # type: ignore[index]
-            elif kwargs["option"] == "0_WMax/10":
-                self.values[:, 0] = 0.0  # type: ignore[index]
-                self.values[:, 1] = 0.1 * con.wmax  # type: ignore[index]
-            elif kwargs["option"] == "FK":
-                self.values[:, 0] = con.fk  # type: ignore[index]
-                self.values[:, 1] = con.fk  # type: ignore[index]
-            else:
-                raise ValueError(
-                    f"Parameter {objecttools.elementphrase(self)} supports "
-                    f"the options `FK`, `0_WMax/10`, and `FK/2_FK`, but "
-                    f"`{kwargs['option']}` is given."
-                ) from None
+        else:
+            self._keywordarguments = parametertools.KeywordArguments(
+                option=kwargs["option"]
+            )
+            self.values = self._get_values_according_to_option()
+
+    def _get_values_according_to_option(self) -> NDArrayFloat:
```

Comment:
NDArrayFloat -> MatrixFloat (a little more precise; will become relevant when we start to type the shape of our Parameter and Sequence_ subclasses.)

Reply:
Done.